packages/drivers/routerlicious-driver/src/gitManager.ts

```diff
@@ -0,0 +1,316 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { assert } from "@fluidframework/common-utils";
+import * as resources from "@fluidframework/gitresources";
+import { buildHierarchy } from "@fluidframework/protocol-base";
+import * as api from "@fluidframework/protocol-definitions";
+import {
+	ICreateRefParamsExternal,
+	IPatchRefParamsExternal,
+	IWholeFlatSummary,
+	IWholeSummaryPayload,
+	IWriteSummaryResponse,
+} from "@fluidframework/server-services-client";
+import { IGitManager, IHistorian } from "./storageContracts";
+import { IR11sResponse } from "./restWrapper";
+
+export class GitManager implements IGitManager {
+	private readonly blobCache = new Map<string, resources.IBlob>();
+	private readonly commitCache = new Map<string, resources.ICommit>();
+	private readonly treeCache = new Map<string, resources.ITree>();
+	private readonly refCache = new Map<string, string>();
+
+	constructor(private readonly historian: IHistorian) {}
+
+	public async getHeader(id: string, sha: string): Promise<api.ISnapshotTree> {
+		const header = await this.historian.getHeader(sha);
+
+		// Cache blobs that were sent in the header
+		for (const blob of header.blobs) {
+			this.blobCache.set(blob.sha, blob);
+		}
+
+		return buildHierarchy(header.tree);
+	}
+
+	public async getFullTree(sha: string): Promise<any> {
+		return this.historian.getFullTree(sha);
+	}
+
+	public async getCommit(sha: string): Promise<resources.ICommit> {
+		const commit = this.commitCache.get(sha);
+		if (commit !== undefined) {
+			return commit;
+		}
+
+		return this.historian.getCommit(sha);
+	}
+
+	/**
+	 * Reads the object with the given ID. We defer to the client implementation to do the actual read.
+	 */
+	public async getCommits(shaOrRef: string, count: number): Promise<resources.ICommitDetails[]> {
```

Comment:
Do we use it? Please go over public APIs of cloned classes and check if all the members are actually used. It's quite possible that the answer is no and we can actually reduce bundle size

Reply:
On surface, write(), createTreeCore() are possible candidates for removal